src/main/java/org/prebid/server/bidder/ttx/TtxBidder.java

```diff
@@ -99,90 +99,88 @@ private ExtImpTtx parseImpExt(Imp imp) {
         }
     }
 
-    private Imp updateFirstImp(Imp firstImp, String productId,
-                               String zoneId, List<BidderError> errors) {
-        final Imp.ImpBuilder modifiedFirstImp = firstImp.toBuilder();
-        modifiedFirstImp.ext(createImpExt(productId, zoneId)).build();
-
-        final Video video = firstImp.getVideo();
-        try {
-            if (video != null) {
-                modifiedFirstImp.video(updateVideo(video, productId));
-            }
-        } catch (PreBidException e) {
-            errors.add(BidderError.badInput(e.getMessage()));
-        }
-
-        return modifiedFirstImp.build();
+    private static Site updateSite(Site site, String siteId) {
+        return site == null ? null : site.toBuilder().id(siteId).build();
     }
 
-    private List<Imp> replaceFirstImp(List<Imp> imps, Imp firstImp) {
-        final List<Imp> updatedImpList = new ArrayList<>(imps);
-        updatedImpList.set(0, firstImp);
-        return updatedImpList;
+    private Imp updateImp(Imp imp, String productId, String zoneId, List<BidderError> errors) {
+
+        return imp.toBuilder()
+                .video(updateVideo(imp.getVideo(), productId, errors))
+                .ext(createImpExt(productId, zoneId))
+                .build();
     }
 
     private ObjectNode createImpExt(String productId, String zoneId) {
         final TtxImpExt ttxImpExt = TtxImpExt.of(TtxImpExtTtx.of(productId, StringUtils.stripToNull(zoneId)));
         return mapper.mapper().valueToTree(ttxImpExt);
     }
 
-    private Video updateVideo(Video video, String productId) {
+    private static Video updateVideo(Video video, String productId, List<BidderError> errors) {
```

Comment:
probably better name it updatedVideo (my mistake)

Just don't really like when we use null to check later if we need to update logic, and try to be more clear with that.